### ISSUE TITLE
Wrong Snapshot Store Output

### ISSUE THREAD

**Issue by helloqiu** [syrusakbary/snapshottest#62]
Hi there. I think I find a bug :(
## Reproduce the Problem
If I simply `assert_match` an `Exception`, the output is **right**:  
``` python
# test_example.py
def test_example(snapshot):
  example = Exception()
  snapshot.assert_match(example)
```
Generated snapshot file:  
``` python
# -*- coding: utf-8 -*-
# snapshottest: v1 - https://goo.gl/zC4yUc
from __future__ import unicode_literals

from snapshottest import GenericRepr, Snapshot


snapshots = Snapshot()

snapshots['test_example 1'] = GenericRepr('Exception()')
```
But the output is **wrong** when I `assert_match` a `dict` which contains this `Exception`:
``` python
# test_example.py
def test_example(snapshot):
  example = dict(err=Exception())
  snapshot.assert_match(example)
```
Generated snapshot file:  
``` python
# -*- coding: utf-8 -*-
# snapshottest: v1 - https://goo.gl/zC4yUc
from __future__ import unicode_literals

from snapshottest import GenericRepr, Snapshot


snapshots = Snapshot()

snapshots['test_example 1'] = {
    'err': Exception()
}
```
The expected result shoud be `'err': GenericRepr('Exception()')` as the first example.  
## Possible Solutions
Maybe this bug is caused here: 
 https://github.com/syrusakbary/snapshottest/blob/master/snapshottest/formatters.py#L20
https://github.com/syrusakbary/snapshottest/blob/master/snapshottest/module.py#L222
In the second example, the `Formatter.get_formatter()` returns a `TypeFormatter` which `store` function return the dict itself directely. It should get the formatter for every key&value in the dict like `format` and parse them recursively.

**Comment by ygormutti** [comment]
Fixed in master by #82. cc @yourbuddyconner

**Comment by yourbuddyconner** [comment]
Closing as Fixed, please reopen if you are still experiencing issues @helloqiu
